src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -370,4 +595,143 @@ mod tests {
+            "CEC348F4EF5F84D3AA6C4FA184C65766",
+        ),
+        (
+            "aes-128-ecb",
+            "pingcap",
+            "123456789012345678901234",
+            "6F1589686860C8E8C7A40A78B25FF2C0",
+        ),
+        (
+            "aes-128-ecb",
+            "pingcap",
+            "123",
+            "996E0CA8688D7AD20819B90B273E01C6",
+        ),
+        (
+            "aes-192-ecb",
+            "pingcap",
+            "1234567890123456",
+            "9B139FD002E6496EA2D5C73A2265E661",
+        ),
+        (
+            "aes-256-ecb",
+            "pingcap",
+            "1234567890123456",
+            "F80DCDEDDBE5663BDB68F74AEDDB8EE3",
+        ),
+    ];
+
+    #[test]
+    fn test_aes_encrypt() {
+        use super::EvalContext;
+        let mut ctx = EvalContext::default();
+        for (aes_name, origin, key, crypt) in DATA {
+            let aes_name = datum_expr(Datum::Bytes(aes_name.as_bytes().to_vec()));
+            let origin = datum_expr(Datum::Bytes(origin.as_bytes().to_vec()));
+            let key = datum_expr(Datum::Bytes(key.as_bytes().to_vec()));
+            let op = scalar_func_expr(ScalarFuncSig::AesEncrypt, &[aes_name, origin, key]);
+            let op = Expression::build(&mut ctx, op).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            let crypt = Datum::Bytes(hex::decode(crypt.as_bytes().to_vec()).unwrap());
+            assert_eq!(got, crypt);
+        }
+    }
+
+    #[test]
+    fn test_aes_decrypt() {
+        use super::EvalContext;
+        let mut ctx = EvalContext::default();
+        for (aes_name, origin, key, crypt) in DATA {
+            let aes_name = datum_expr(Datum::Bytes(aes_name.as_bytes().to_vec()));
+            let crypt = datum_expr(Datum::Bytes(
+                hex::decode(crypt.as_bytes().to_vec()).unwrap(),
+            ));
+            let key = datum_expr(Datum::Bytes(key.as_bytes().to_vec()));
+            let op = scalar_func_expr(ScalarFuncSig::AesDecrypt, &[aes_name, crypt, key]);
+            let op = Expression::build(&mut ctx, op).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            let origin = Datum::Bytes(origin.as_bytes().to_vec());
+            assert_eq!(got, origin);
+        }
+    }
+
+    #[test]
+    fn test_aes_null() {
+        use super::EvalContext;
+        let arg_null = datum_expr(Datum::Bytes(b"".to_vec()));
+        let mut ctx = EvalContext::default();
+        let aes_name = datum_expr(Datum::Bytes(b"aes-128-ecb".to_vec()));
+        let crypt = datum_expr(Datum::Bytes(
+            hex::decode(b"697BFE9B3F8C2F289DD82C88C7BC95C4".to_vec()).unwrap(),
+        ));
+        let op = scalar_func_expr(ScalarFuncSig::AesDecrypt, &[aes_name, crypt, arg_null]);
+        let op = Expression::build(&mut ctx, op).unwrap();
+        let got = op.eval(&mut ctx, &[]).unwrap();
+        assert_eq!(got, Datum::Bytes(b"".to_vec()));
+
+        let arg_null = datum_expr(Datum::Bytes(b"".to_vec()));
```

Comment:
Actually, it is empty string arg instead of null arg. You can use Datum::Null for null arg.

Reply:
Ping :)